core/src/main/java/org/apache/accumulo/core/constraints/DefaultKeySizeConstraint.java

```diff
@@ -24,10 +24,19 @@
 import org.apache.accumulo.core.data.ColumnUpdate;
 import org.apache.accumulo.core.data.Mutation;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
 /**
  * A constraints that limits the size of keys to 1mb.
+ *
+ * @deprecated since 2.1.0 Use
+ *             {@link org.apache.accumulo.core.data.constraints.DefaultKeySizeConstraint}
  */
-public class DefaultKeySizeConstraint implements Constraint {
+@Deprecated(since = "2.1.0")
+@SuppressFBWarnings(value = "NM_SAME_SIMPLE_NAME_AS_SUPERCLASS",
+    justification = "Same name used for compatibility during deprecation cycle")
+public class DefaultKeySizeConstraint
```

Comment:
There are two possible options, one is to extend and make the class final like the following.  This supports configuring the class, but does not support anyone extending the class in java.  This would break any class extending the class.

```java
public final class DefaultKeySizeConstraint  extends org.apache.accumulo.core.data.constraints.DefaultKeySizeConstraint {
  // intentionally empty, only want to support configuring this class
}
```

I think the best way to support configuring and extending is w/ a copy of the existing code AND not extending the other DefaultKeySizeConstraint.

```java
// implement the interface in the current package AND deprecate
public class DefaultKeySizeConstraint implements Constraint {
  // carry forward the existing code (before this commit) as-is
}
```

It only seems the 2nd option may be worthwhile if someone could change maxSize in a subclass.  I am not sure that is possible since its final, but I have never tried to override something that is final and protected in a subclass.